src/main/scala/ai/privado/languageEngine/java/tagger/collection/MethodFullNameCollectionTagger.scala

```diff
@@ -0,0 +1,148 @@
+    builder: DiffGraphBuilder,
+    collectionPoints: List[AstNode],
+    collectionRuleInfo: RuleInfo,
+    returnByName: Boolean = false
+  ): Unit = {
+    collectionPoints.foreach(collectionPoint => {
+      addRuleTags(builder, collectionPoint, collectionRuleInfo, ruleCache)
+      storeForTag(builder, collectionPoint, ruleCache)(
+        InternalTag.COLLECTION_METHOD_ENDPOINT.toString,
+        getFinalEndPoint(collectionPoint, returnByName)
+      )
+    })
+  }
+
+  private def getFinalEndPoint(collectionPoint: AstNode, returnByName: Boolean): String = {
+    if (returnByName && collectionPoint.isMethod) {
+      collectionPoint.asInstanceOf[Method].name
+    } else {
+      val methodUrl = methodUrlMap.getOrElse(collectionPoint.id(), "")
+      Try(classUrlMap.getOrElse(collectionPoint.asInstanceOf[Method].typeDecl.head.id(), "")) match {
+        case Success(classUrl) => classUrl + methodUrl
+        case Failure(e) =>
+          methodUrl
+      }
+    }
+  }
+
+  private def collectUrlsFromHandlerEndpoints(combinedRulePatterns: String): Map[Long, Method] = {
+    val methodCalls       = cpg.call.methodFullName(combinedRulePatterns).l
+    val methods           = scala.collection.mutable.HashMap.empty[Long, Method]
+    val localMethodUrlMap = scala.collection.mutable.HashMap.empty[Long, String]
+    for (methodCall <- methodCalls) {
+      val url                           = methodCall.argument.isLiteral.code.head
+      var handlerMethod: Option[Method] = Option.empty[Method]
+
+      breakable {
+        if (methodCall.argument.length < 2) { // we do not have enough arguments to get the handler method
+          break                               // empty handler, continue loop
+        }
+        // match on second argument, that's the handler
+        methodCall.argument(2) match {
+          case c: Call =>
+            if (c.methodFullName == "<operator>.fieldAccess") { // E.g. AnotherHandlerClass.someHandler
+              val children = c.astChildren.l
+              // Anything but a field identifier is the method for us, Call, TypeRef, TypeDecl.
+              val fieldMap = children.filter(!_.isFieldIdentifier).head.toMap
+              if (fieldMap.isEmpty) {
+                break
+              }
+              val classNameOption: Option[Any] = fieldMap.get("TYPE_FULL_NAME")
+              if (classNameOption.isEmpty) {
+                break
+              }
+              // Kotlin has a '$' in this typename, so getting everything before that
+              val className  = classNameOption.get.toString.split('$')(0)
+              val methodName = children.isFieldIdentifier.canonicalName.headOption
+              if (className != "" && methodName.isDefined) {
+                // get the assignment and the handler from there
+                /** this.endpointHandler = { req: Request, res: Response -> "hello from handler" }
+                  */
+                handlerMethod = cpg.file
```

Comment:
This is the complicated way to get the actual method handler reference for tagging.